src/NHibernate/Impl/FilterImpl.cs

```diff
@@ -93,10 +93,19 @@ public IFilter SetParameterList(string name, ICollection values)
 			if (values.Count > 0)
 			{
 				var e = values.GetEnumerator();
-				e.MoveNext();
-				if (!type.ReturnedClass.IsInstanceOfType(e.Current))
+				try
```

Comment:
Both `FirstOrDefault` and `Any` would change the exact technical semantic of this code. It is currently "if there are any elements (`values.Count > 0`), take the first and check it has an appropriate type". `FirstOrDefault` does not let us know whether there was no element or there was one but the it was the default one. I have not put a `First` here because I should have issue a `Cast<object>()` before for using `First`.

Now if we change that to a test of all supplied values, not only of the first one if any, why not using `Cast<object>().Any(...)`.